apps/ledger-live-mobile/src/components/BleDevicePairingFlow/index.tsx

```diff
@@ -91,22 +94,35 @@ const BleDevicePairingFlow: React.FC<BleDevicePairingFlowProps> = ({
   const [pairingFlowStep, setPairingFlowStep] = useState<PairingFlowStep>("scanning");
 
   const [deviceToPair, setDeviceToPair] = useState<Device | null>(null);
+  const [deviceLocked, setDeviceLocked] = useState<Device | null>(null);
   const [isPaired, setIsPaired] = useState(false);
 
-  const onDeviceSelect = useCallback(
-    (item: Device) => {
-      const deviceToPair = {
-        deviceId: item.deviceId,
-        deviceName: item.deviceName,
-        modelId: item.modelId,
-        wired: false,
-      };
+  const { pairingError } = useBleDevicePairing({
+    deviceId: deviceToPair?.deviceId,
+  });
```

Comment:
What scares me is that now we trigger the pairing flow with `useBleDevicePairing` in both  `apps/ledger-live-mobile/src/components/BleDevicePairingFlow/index.tsx` and `apps/ledger-live-mobile/src/components/BleDevicePairingFlow/BleDevicePairing.tsx`

I think this is a bad idea :/ Why can't you put `UnlockDeviceDrawer` in `BleDevicePairing` ? And if it's a problem, maybe we could give to `BleDevicePairing` a setState prop like `setIsDeviceLocked` and have this `isDeviceLocked` state live here ;)

Reply:
Yes this might not be the best solution indeed.

I found a way to keep the hook on `BleDevicePairing.tsx` and the drawer on `index.tsx`by calling a `onLock` callback from `index.tsx` to display the drawer.

The challenge was to not display the generic error from `BleDevicePairing.tsx` at the same time, so with the callback it redirect to previous screen, then popup the lock drawer without displaying any errors.